src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_CollectionExpression.cs

```diff
@@ -159,16 +168,35 @@ private BoundExpression VisitCollectionBuilderCollectionExpression(BoundCollecti
             Debug.Assert(!_inExpressionLambda);
             Debug.Assert(node.Type is { });
 
+            var syntax = node.Syntax;
+            var elements = node.Elements;
             var constructMethod = node.CollectionBuilderMethod;
+
             Debug.Assert(constructMethod is { });
             Debug.Assert(constructMethod.ReturnType.Equals(node.Type, TypeCompareKind.AllIgnoreOptions));
 
             var spanType = (NamedTypeSymbol)constructMethod.Parameters[0].Type;
             Debug.Assert(spanType.OriginalDefinition.Equals(_compilation.GetWellKnownType(WellKnownType.System_ReadOnlySpan_T), TypeCompareKind.AllIgnoreOptions));
 
-            var span = VisitArrayOrSpanCollectionExpression(node, spanType, spanType.TypeArgumentsWithAnnotationsNoUseSiteDiagnostics[0].Type);
-            return new BoundCall(
-                node.Syntax,
+            var elementType = spanType.TypeArgumentsWithAnnotationsNoUseSiteDiagnostics[0];
+            var locals = ArrayBuilder<LocalSymbol>.GetInstance();
+            var sideEffects = ArrayBuilder<BoundExpression>.GetInstance();
+            BoundExpression span;
+
+            if (elements.Length > 0
+                && !elements.Any(i => i is BoundCollectionExpressionSpreadElement)
+                && _compilation.Assembly.RuntimeSupportsInlineArrayTypes
+                && (!constructMethod.ReturnType.IsRefLikeType || constructMethod.Parameters[0].EffectiveScope == ScopedKind.ScopedValue))
```

Comment:
Did we add tests to cover those conditions, or were those getting hit by existing scenarios?

Reply:
Covered by `CollectionBuilder_RefStructCollection`.